mergify_engine/actions/merge/action.py

```diff
@@ -35,6 +41,45 @@
 MARKDOWN_COMMIT_MESSAGE_RE = re.compile(r"^#+ Commit Message ?:?\s*$", re.I)
 
 
+@dataclasses.dataclass
+class UndefinedVariables(Exception):
+    variables: set
+
+
+class PullRequestContext(jinja2.runtime.Context):
+    """This is a special Context that resolves any attribute first in the "pull request" object.
+
+
+    This allows to write {{author}} instead of {{pull_request.author}}."""
```

Comment:
If you do that it means it means it will request all values for all items of the pull request which makes it do extra HTTP requests to get e.g. the list of checks, etc.

Reply:
Oh, nice!